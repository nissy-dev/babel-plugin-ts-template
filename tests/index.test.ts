import pluginTester from "babel-plugin-tester";
import myPlugin from "../src/index";

pluginTester({
  plugin: myPlugin,
  babelOptions: {
    parserOpts: {
      // If you want to parse typescript or jsx, these plugins are required
      plugins: ["jsx", "typescript"],
    },
  },
  filename: __filename,
  pluginOptions: {
    option1: true,
  },
  tests: {
    "basic test": {
      code: 'var a = "test";',
      output: 'const a = "test";',
    },
    "basic test using files": {
      fixture: "__fixtures__/sample.js",
      outputFixture: "__fixtures__/sample-out.js",
    },
    "snapshot test": {
      code: 'var a = "test";',
      snapshot: true,
    },
  },
});
