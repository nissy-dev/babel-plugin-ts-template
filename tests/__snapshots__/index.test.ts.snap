// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin snapshot test: snapshot test 1`] = `

var a = "test";

      ↓ ↓ ↓ ↓ ↓ ↓

const a = "test";


`;
